src/circuit/gadget/ecc/chip/mul/incomplete.rs

```diff
@@ -0,0 +1,296 @@
+            let y_p_check = y_p_cur.clone() - y_p_next;
+
+            // λ_{1,i}⋅(x_{A,i} − x_{P,i}) − y_{A,i} + (2k_i - 1) y_{P,i} = 0
+            let poly1 = lambda1_cur.clone() * (x_a_cur.clone() - x_p_cur.clone()) - y_a_cur.clone()
+                + (k * pallas::Base::from_u64(2) + Expression::Constant(-pallas::Base::one()))
+                    * y_p_cur;
+
+            // (λ_{1,i} + λ_{2,i})⋅(x_{A,i} − (λ_{1,i}^2 − x_{A,i} − x_{P,i})) − 2y_{A,i}) = 0
+            let poly2 = {
+                let lambda_neg = lambda1_cur.clone() + lambda2_cur.clone();
+                let lambda1_expr =
+                    lambda1_cur.clone() * lambda1_cur.clone() - x_a_cur.clone() - x_p_cur.clone();
+                lambda_neg * (x_a_cur.clone() - lambda1_expr)
+                    - Expression::Constant(pallas::Base::from_u64(2)) * y_a_cur.clone()
+            };
+
+            // λ_{2,i}^2 − x_{A,i+1} −(λ_{1,i}^2 − x_{A,i} − x_{P,i}) − x_{A,i} = 0
+            let poly3 = lambda2_cur.clone() * lambda2_cur.clone()
+                - x_a_next.clone()
+                - (lambda1_cur.clone() * lambda1_cur)
+                + x_p_cur;
+
+            // λ_{2,i}⋅(x_{A,i} − x_{A,i+1}) − y_{A,i} − y_{A,i+1} = 0
+            let poly4 = lambda2_cur * (x_a_cur - x_a_next) - y_a_cur - y_a_next;
+
+            array::IntoIter::new([bool_check, x_p_check, y_p_check, poly1, poly2, poly3, poly4])
+                .map(move |poly| q_mul.clone() * poly)
+        });
+    }
+
+    // We perform incomplete addition on all but the last three bits of the
+    // decomposed scalar.
+    // We split the bits in the incomplete addition range into "hi" and "lo"
+    // halves and process them side by side, using the same rows but with
+    // non-overlapping columns.
+    // Returns (x, y, z).
+    #[allow(clippy::type_complexity)]
+    pub(super) fn double_and_add(
+        &self,
+        region: &mut Region<'_, pallas::Base>,
+        offset: usize,
+        base: &EccPoint,
+        bits: &[Option<bool>],
+        acc: (X<pallas::Base>, Y<pallas::Base>, Z<pallas::Base>),
+    ) -> Result<(X<pallas::Base>, Y<pallas::Base>, Vec<Z<pallas::Base>>), Error> {
+        // Check that we have the correct number of bits for this double-and-add.
+        assert_eq!(bits.len(), self.num_bits);
+
+        // Handle exceptional cases
+        let (x_p, y_p) = (base.x.value(), base.y.value());
+        let (x_a, y_a) = (acc.0.value(), acc.1 .0);
+
+        if let (Some(x_a), Some(y_a), Some(x_p), Some(y_p)) = (x_a, y_a, x_p, y_p) {
+            // A is point at infinity
+            if (x_p == pallas::Base::zero() && y_p == pallas::Base::zero())
+            // Q is point at infinity
+            || (x_a == pallas::Base::zero() && y_a == pallas::Base::zero())
+            // x_p = x_a
+            || (x_p == x_a)
+            {
+                return Err(Error::SynthesisError);
+            }
+        }
+
+        // Enable `q_mul` on all but the last row of the incomplete range.
```

Comment:
Yep, disabling this on the last row means we aren't constraining the `y_A` output of this `double_and_add`.